### ISSUE TITLE
BUG : the callback function is not called, when do a quick click on the buttons immediately after opening of the popup 

### ISSUE THREAD

**Issue by nicolomonili** [t4t5/sweetalert#281]
As the title suggests, the problem occurs (in this case), when I press the "OK" button, just after the opening of the popup. If pressed immediately after the opening of the popup, the callback function is not called.

[JSFiddle example](http://jsfiddle.net/xe096w10/7/)

```
<button id="btnShowAlert">
    Show Sweet Alert
</button>

$("#btnShowAlert").click(function() {

    swal({
        title : "Test",
        text : "Test",
        type : "success"
    },function(){
        alert("confirm");
    });

});
```

**Comment by nicolomonili** [comment]
No update ?

**Comment by nicolomonili** [comment]
+1

**Comment by nicolomonili** [comment]
+1

**Comment by SEAPUNK** [comment]
+1ing this, happens to me too.

**Comment by nicolomonili** [comment]
For now i put this at the end of the function "**var openModal = function openModal(callback) {..**" : 

    setTimeout(function(e){
  	    swal.disableButtons();
  	    setTimeout(function(dddd){
  		   swal.enableButtons();
  	    }, 500);
     }, 0);

**Comment by jhovgaard** [comment]
+1

**Comment by drassi** [comment]
FYI folks, the [limonte/sweetalert2 fork](https://github.com/limonte/sweetalert2) fixes this issue for me, and requires minimal changes to implement.

**Comment by t4t5** [comment]
This is now fixed in 2.0, as there's no longer any delay. Please check out the [upgrade guide](https://sweetalert.js.org/guides/#upgrading-from-1x)

**Comment by pnstech** [comment]
Happens with me too
